assets/src/edit-story/components/library/panes/text/karma/textPane.karma.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { Fixture } from '../../../../../karma/fixture';
+
+describe('CUJ: Creator can Add and Write Text: Consecutive text presets', () => {
+  let fixture;
+  beforeEach(async () => {
+    fixture = new Fixture();
+    await fixture.render();
+  });
+
+  afterEach(() => {
+    fixture.restore();
+  });
+
+  it('should add text presets below each other if added consecutively', async () => {
+    await fixture.editor.library.textTab.click();
+    await fixture.events.click(fixture.editor.library.text.preset('Heading 1'));
+    await fixture.events.click(fixture.editor.library.text.preset('Heading 3'));
+    await fixture.events.click(fixture.editor.library.text.preset('Paragraph'));
+
+    await fixture.snapshot('consecutively added different text presets');
+  });
+
+  it('should ensure staggered presets fit on the page', async () => {
+    await fixture.editor.library.textTab.click();
+    // Stagger all different text presets.
+    await fixture.events.click(fixture.editor.library.text.preset('Heading 1'));
+    await fixture.events.click(fixture.editor.library.text.preset('Paragraph'));
+    await fixture.events.click(fixture.editor.library.text.preset('Heading 2'));
+    await fixture.events.click(fixture.editor.library.text.preset('Paragraph'));
+    await fixture.events.click(fixture.editor.library.text.preset('Heading 3'));
+    await fixture.events.click(fixture.editor.library.text.preset('Caption'));
+    await fixture.events.click(fixture.editor.library.text.preset('Paragraph'));
+    await fixture.events.click(fixture.editor.library.text.preset('OVERLINE'));
```

Comment:
Hm, [there's something fishy in the screenshot though](https://percy.io/google/web-stories-wp/builds/6671293/view/382333224?browser=chrome&mode=head&width=1600). _Overline_ is added to the bottom, while the _paragraph_ wrapped up at the original starting point. _Overline_ added last should have gone right below the last _paragraph_ (added at the original offset).

I think the history didn't manage to update between these, which is why this can occur. It's not super critical but might make the test a bit flaky.

Maybe you should create a wrapper method that makes sure to wait for history to update before proceeding with the next text insertion.

WDYT?

Reply:
Hmm, looks like the issue is with the first "Heading 1" and the first "Paragraph" -- it looks like the first paragraph was added on top of Heading 1 instead of below it -- the following elements seem to be staggered correctly after that.

Indeed it looks like the history didn't update properly meanwhile -- seems to be an issue with the test only.